### PR TITLE
Marcos.cifuentes/feature card list fragment service

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
In this PR I added the Service needed to do a request to the API and recover the Pokemon Card List from a given Type/Supertype/Subtype element selected, removing the need to use a mocked list.

- Built the structure of the request.
- Modified the mocked classes to use the Service response.
- Modified unit tests for the implementation.

Tests passing and coverage:

cardList service tests passed

cardList service tests coverage

GIF of the app:

pokemon card list service

### CODE REVIEW

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/java/com/globant/pokemontcgapp/activity/PokemonCardListActivity.kt [hunk 1]
```diff
@@ -23,36 +25,43 @@ class PokemonCardListActivity : AppCompatActivity() {
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityPokemonCardListBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        val selection = intent.getStringExtra(SELECTION)
-        val selectionColor = intent.getIntExtra(SELECTION_COLOR, DEFAULT_VALUE)
-        initUi(selection, selectionColor)
+        group = intent.getStringExtra(POKEMON_GROUP)
+        groupSelected = intent.getStringExtra(SELECTION)
```
Comment (nicolbidone) [inline app/src/main/java/com/globant/pokemontcgapp/activity/PokemonCardListActivity.kt]: can you give group vars a more descriptive name?

File: domain/src/main/java/com/globant/domain/entity/PokemonCard.kt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.globant.domain.entity
 
 class PokemonCard(
-    var id: Int,
+    var id: String,
```
Comment (nicolbidone) [inline domain/src/main/java/com/globant/domain/entity/PokemonCard.kt:4]: why the id's are strings?

File: domain/src/main/java/com/globant/domain/entity/PokemonCardResponse.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.globant.domain.entity
+
+class PokemonCardResponse(
+    var id: String,
```
Comment (nicolbidone) [inline domain/src/main/java/com/globant/domain/entity/PokemonCardResponse.kt:4]: same

File: app/src/main/java/com/globant/pokemontcgapp/activity/PokemonCardListActivity.kt (hunk 1 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/activity/PokemonCardListActivity.kt]: Refactored for a more descriptive name, thanks for the help

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**